### PR TITLE
Redis, RQ, background-worker

### LINKED ISSUE #10: Create a python service template repository
## Problem

TogetherCrew is based on micro-services. As the number of micro-services will continue to grow, it is of vital importance to ensure that our services use the same pipelines, structures, and libraries (where possible).

## Solution

We will create a service template for python services. Similar to the typescript template, developers should be able to use it as a basis for any service, saving days in configuring a new service.

It should support the following commands:

### Linter
```
docker run -e RUN_LOCAL=true -e TYPESCRIPT_DEFAULT_STYLE=prettier -e VALIDATE_DOCKERFILE_HADOLINT=false -v $(pwd):/tmp/lint github/super-linter:slim-latest
```

### Tests
```
docker compose -f docker-compose.test.yml up --exit-code-from app --build
```

### Development Environment
```
docker compose -f docker-compose.dev.yml up
```

## Tasks

- [x] PST: Create Dockerfile
- [x] PST: Create .dockerignore
- [x] PST: Create docker-compose.dev.yml
- [x] PST: Create docker-compose.test.yml
- [x] PST: Create .gitignore 
- [x] PST: Github action implementation
- [x] PST: Implement RQ for background tasks
- [x] PST: Align codebase to validate super-linter
- [x] PST: Neo4j connection
- [x] PST: MongoDB connection
- [x] PST: RabbitMQ connection
- [x] PST: Redis connection
- [x] PST: Background Worker

## Documentation

- https://python-rq.org/
- https://github.com/RnDAO/tc-operation/tree/main/tc-DAOlytics-operation

## Notes

Most of the code can be ported from https://github.com/RnDAO/tc-operation/tree/main/tc-DAOlytics-operation

## Overview

The goal of the python service template is to allow us to easily create new datasource analyzers. The analysis will occur in the background using workers. To accomplish this, we will use https://python-rq.org/.

### Server

The server is responsible for the following:
- Listening for messages on RabbitMQ
- Enqueuing jobs onto a queue
- Acknowledging that a message was received

### Worker

Once a job is enqueued, it is the Workers responsibility to process the job.
- Handle the connect to datasource i.e. Neo4j, MongoDB
- Execute custom logic
- Emit messages to RabbitMQ

#### Datasources
- The developer should have then option to enable or disable a certain datasource. This can be done by simply commenting code.
- The RabbitMQ connection should be managed by (using https://github.com/TogetherCrew/broker-lib-py)

## Test cases
- Should listen to messages from RabbitMQ
- Should process job (hello world)
- Should emit messages to RabbitMQ

### PR DESCRIPTION
**Pull request by scientiststwin**

### CODE REVIEW

**Review by amindadgar — APPROVED** [review]

File: redis_worker.py
Comment (cyri113) [inline redis_worker.py:1]: why are we calling it `redis_worker.py` and not just `worker.py`

File: redis_worker.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+def worker_exception_handler(job, exc_type, exc_value, traceback):
+    logging.error(" ========= RQ Exception =========")
+    logging.error(f"JOB: {job}")
+    logging.error(f"exc_type: {exc_type}")
+    logging.error(f"exc_value: {exc_value}")
+    logging.error(f"traceback: {traceback}")
+
+
+if __name__ == "__main__":
+    host = os.getenv("REDIS_HOST", "localhost")
```
Comment (cyri113) [inline redis_worker.py]: using defaults for this isn't great practice, it makes devs lazy and can create issues further down the line.

It should through an error if there isn't a `REDIS_HOST`

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,6 @@
+async-timeout==4.0.2
 attrs==22.2.0
+click==8.1.6
```
Comment (cyri113) [inline requirements.txt:3]: what is `click` used for?

**Review by cyri113 — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 2 above)
Comment (scientiststwin) [reply requirements.txt:3]: I think It was added with the `rq` package. It was not installed directly

File: redis_worker.py (hunk 1 above)
Comment (scientiststwin) [reply redis_worker.py]: right, I was thinking like you! but super-linter forced us to make a default value for the `host` variable. 
It was added in the last commit :)

File: redis_worker.py
Comment (scientiststwin) [reply redis_worker.py:1]: Okay, I'll change the name.

File: redis_worker.py (hunk 1 above)
Comment (cyrillederche) [reply redis_worker.py]: what about nil / null ? will this raise an error?

**Review by cyri113 — APPROVED** [review]

**Pull request merged**